kubernetes_deploy/development/deployment.yml

```diff
@@ -5,6 +5,8 @@ metadata:
     kubernetes.io/change-cause: "<to be filled in deploy_to_kubernetes script>"
   name: laa-legal-adviser-api
 spec:
+  # This pod contains the app, database and rabbitmq containers. Keep this count to 1 otherwise you will have
+  # multiple rabbitmq instances, alternatively you can create a pod for each of these containers
```

Comment:
In hindsight, it would make more sense to create separate pods.

To quote from [Kubernetes in Action](https://www.manning.com/books/kubernetes-in-action):

> [...] when deciding whether to put two containers into a single pod or into two separate pods, you always need to ask yourself the following questions:
>
> * Do they need to be run together or can they run on different hosts?
> * Do they represent a single whole or are they independent components?
> * Must they be scaled together or individually?

Reply:
if we do do that, then we need the following added to the .circleci/deploy_to_kubernetes script
`if [ -f "$NAMESPACE_DIR/amq_deployment.yml" ] ; then
# Worker queue deployment
cat "$NAMESPACE_DIR/amq_deployment.yml" | \
  kubectl set image worker="$ECR_DEPLOY_IMAGE" --filename=/dev/stdin --local --output=yaml | \
  kubectl annotate kubernetes.io/change-cause="$CIRCLE_BUILD_URL" --filename=/dev/stdin --local --output=yaml | \
  kubectl apply --record=false \
    --filename=/dev/stdin
fi

if [ -f "$NAMESPACE_DIR/postgress_deployment.yml" ] ; then
# Worker queue deployment
cat "$NAMESPACE_DIR/postgress_deployment.yml" | \
  kubectl set image worker="$ECR_DEPLOY_IMAGE" --filename=/dev/stdin --local --output=yaml | \
  kubectl annotate kubernetes.io/change-cause="$CIRCLE_BUILD_URL" --filename=/dev/stdin --local --output=yaml | \
  kubectl apply --record=false \
    --filename=/dev/stdin
fi
`

The conditional statement is required because that deployment script is shared and those deployments will not be used in staging/prod

Reply:
Not necessarily as we already have a `docker/local_deploy.sh` script that deals with local deployment, so local deployment specific extras could go there. These files are possibly fine without any substitution so they will most likely be `kubectl apply --filename=kubernetes_deploy/development/postgres_deployment.yaml` or similar.

Reply:
Ah that's a really good idea. I added this change

Reply:
With the changes to services, we no longer need these comments. 🎉